### PR TITLE
Create test for workers and switch to Python 3.6

### PR DESCRIPTION
**Pull request by TomasHodor**
Signed-off-by: Tomas <thodor@frinx.io>

### CODE REVIEW

File: microservices/netinfra_utils/workers/netconf_worker_test.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+            self.assertEqual(request["output"]["response_code"], 201)
+            self.assertEqual(request["output"]["response_body"], {})
+
+    def test_mount_existing_device(self):
+        with patch('netconf_worker.requests.put') as mock:
+            mock.return_value = MockResponse(bytes(json.dumps({}), encoding='utf-8'), 204)
+            request = netconf_worker.execute_mount_netconf(
+                {"inputData": {"device_id": "xr6", "host": "192.168.1.1", "port": "830", "keepalive-delay": "1000",
+                               "tcp-only": "false", "username": "name", "password": "password"}})
+            self.assertEqual(request["status"], "FAILED")
```
Comment (MartinSunal) [inline microservices/netinfra_utils/workers/netconf_worker_test.py]: why the status is FAILED? 204 means that data was replaced in uniconfig.

File: microservices/netinfra_utils/workers/netconf_worker_test.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+
+class TestCheckNetconfIdAvailable(unittest.TestCase):
+    def test_execute_check_netconf_id_available_exist(self):
+        with patch('netconf_worker.requests.get') as mock:
+            mock.return_value = MockResponse(bytes(json.dumps(netconf_node_connected), encoding='utf-8'), 200)
+            request = netconf_worker.execute_check_netconf_id_available({"inputData": {"device_id": "xr6"}})
+            self.assertEqual(request["status"], "FAILED")
+            self.assertEqual(request["output"]["url"], odl_url_base
+                             + "/data/network-topology:network-topology/topology=topology-netconf/node=xr6")
```
Comment (MartinSunal) [inline microservices/netinfra_utils/workers/netconf_worker_test.py]: request URL should contain ?content=config

File: microservices/netinfra_utils/workers/netconf_worker_test.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+    def test_execute_check_netconf_id_available_non_exist(self):
+        with patch('netconf_worker.requests.get') as mock:
+            mock.return_value = MockResponse(bytes(json.dumps(netconf_node_non_exist), encoding='utf-8'), 404)
+            request = netconf_worker.execute_check_netconf_id_available({"inputData": {"device_id": "xr6"}})
+            self.assertEqual(request["status"], "COMPLETED")
+            self.assertEqual(request["output"]["url"], odl_url_base
+                             + "/data/network-topology:network-topology/topology=topology-netconf/node=xr6")
```
Comment (MartinSunal) [inline microservices/netinfra_utils/workers/netconf_worker_test.py]: request URL should contain ?content=config

File: microservices/netinfra_utils/workers/cli_worker.py [hunk 4]
```diff
@@ -55,7 +55,7 @@ def execute_mount_cli(task):
     r = requests.put(id_url, data=json.dumps(mount_body), headers=odl_headers, auth=odl_credentials)
     response_code, response_json = parse_response(r)
 
-    if response_code == requests.codes.created or response_code == requests.codes.ok:
+    if response_code == requests.codes.created:
```
[the remark this answers is not in the document]
Comment (TomasHodor) [reply microservices/netinfra_utils/workers/cli_worker.py]: if the device is already mounted, we want to push to the users to unmount the device then mount the device again. Respone with this scenario is always requests.codes.created

File: microservices/netinfra_utils/workers/cli_worker.py [hunk 5]
```diff
@@ -206,28 +206,32 @@ def read_all_devices(url):
...
+    subworkflow = task['inputData']['task']
+    optional = task['inputData']['optional'] if 'optional' in task['inputData'] else "false"
+    add_params = task['inputData']['task_params']
+    add_params = json.loads(add_params) if isinstance(add_params, str) else (add_params if add_params else {})
 
     response_code, response_json = read_all_devices(odl_url_unified_oper_shallow)
 
     if response_code == requests.codes.ok:
-        ids = map(lambda x: x["node-id"], response_json["topology"][0]["node"])
+        print(response_json)
```
[the remark this answers is not in the document]
Comment (TomasHodor) [reply microservices/netinfra_utils/workers/cli_worker.py]: I will remove it and yes i think they are for debug purposes

File: microservices/netinfra_utils/workers/netconf_worker_test.py (hunk 1 above)
Comment (TomasHodor) [reply microservices/netinfra_utils/workers/netconf_worker_test.py]: fixed

File: microservices/netinfra_utils/workers/cli_worker.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (TomasHodor) [reply microservices/netinfra_utils/workers/cli_worker.py]: fixed after discussion

File: microservices/netinfra_utils/workers/netconf_worker_test.py (hunk 2 above)
Comment (TomasHodor) [reply microservices/netinfra_utils/workers/netconf_worker_test.py]: fixed

**Pull request merged**